File: raya/toast/ticket.py
TICKET_PREFIX = "RY"


def guest_ticket_from_toast_order(order: dict | None) -> str | None:
    if not order or not str(order.get("guid") or "").strip():
        return None
    checks = order.get("checks") or []
    if not checks:
        return None
    raw = checks[0].get("displayNumber")
    if raw is None:
        return None
    digits = str(raw).strip()
    if digits.upper().startswith("RY-"):
        digits = digits[3:]
    elif digits.upper().startswith("RY"):
        digits = digits[2:]
    digits = digits.lstrip("#").strip()
    if not digits:
        return None
    return f"{TICKET_PREFIX}-{digits}"

File: src/lib/toast/ticket.ts
import type { ToastOrder } from "@/lib/toast/types";

const TICKET_PREFIX = "RY";

export function guestTicketFromToastOrder(order: ToastOrder | null | undefined): string | null {
  if (!order?.guid?.trim()) return null;

  const raw = order.checks?.[0]?.displayNumber;
  if (raw === undefined || raw === null) return null;

  const digits = String(raw)
    .trim()
    .replace(/^RY-?/i, "")
    .replace(/^#+/, "");

  if (!digits) return null;

  return `${TICKET_PREFIX}-${digits}`;
}
